### PR TITLE
[Merged by Bors] - feat: Set explicit resources on all containers

### PR DESCRIPTION
**Pull request by razvan**
# Description

Part of: https://github.com/stackabletech/issues/issues/394

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ All notable changes to this project will be documented in this file.
 - Generate OLM bundle for Release 23.4.0 ([#442]).
 - Missing CRD defaults for `status.conditions` field ([#443]).
 - Support for OPA 0.51.0 ([#451]).
+- Add support for resource quotas ([#453]).
```
Comment (razvan) [inline CHANGELOG.md:12]:
```suggestion
- Set explicit resources on all containers ([#453]).
```

File: docs/modules/opa/pages/usage.adoc [hunk 2]
```diff
@@ -158,10 +158,10 @@ servers:
       config:
         resources:
           cpu:
-            min: '200m'
-            max: "2"
+            min: 250m
+            max: 500m
           memory:
-            limit: '2Gi'
+            limit: 512Mi
```
Comment (razvan) [inline docs/modules/opa/pages/usage.adoc:164]: I believe we also wanted to document the minimum of resources required to run a simple installation.

Here is an example: https://github.com/stackabletech/druid-operator/pull/444/files#diff-742ddb0f4a8446472c6e339a5f85170cf530742b3ca84072593cea5bddffb7c8

**Review by razvan — COMMENTED** [review]
lgtm! waiting for the tests

File: rust/crd/src/lib.rs [hunk 3]
```diff
@@ -175,11 +175,11 @@ impl OpaConfig {
             logging: product_logging::spec::default_logging(),
             resources: ResourcesFragment {
                 cpu: CpuLimitsFragment {
-                    min: Some(Quantity("200m".to_owned())),
-                    max: Some(Quantity("2".to_owned())),
+                    min: Some(Quantity("250m".to_owned())),
+                    max: Some(Quantity("500".to_owned())),
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:179]: typo

**Review by sbernauer — CHANGES_REQUESTED** [review]
I have added a OPA tab  to google sheets with some resource recommendations. Let's please agree on them first, so that you don't need to edit your PR multiple times

File: docs/modules/opa/pages/usage.adoc (hunk 2 above)
Comment (sbernauer) [reply docs/modules/opa/pages/usage.adoc:164]: Watch out! We need the <whatever> resources for every Node of the clusters (as it's a daemonset)
Special text needed

Comment (maltesander) [reply docs/modules/opa/pages/usage.adoc]: https://github.com/stackabletech/opa-operator/pull/453/commits/116ba6016153d20a397d7ba0e17411046ea3be5a

File: rust/crd/src/lib.rs (hunk 3 above)
Comment (maltesander) [reply rust/crd/src/lib.rs]: done thanks https://github.com/stackabletech/opa-operator/pull/453/commits/6ded35b058bc37634f7705b0131928402ee7ad82

File: rust/operator-binary/src/controller.rs [hunk 4]
```diff
@@ -551,7 +551,8 @@ fn build_server_rolegroup_daemonset(
         )
         .join(" && ")])
         .add_volume_mount(BUNDLES_VOLUME_NAME, BUNDLES_DIR)
-        .add_volume_mount(LOG_VOLUME_NAME, LOG_DIR);
+        .add_volume_mount(LOG_VOLUME_NAME, LOG_DIR)
+        .resources(merged_config.resources.to_owned().into());
```
Comment (sbernauer) [inline rust/operator-binary/src/controller.rs:555]: This is different than what the other operators are doing ;)
Fine for me anyway

File: tests/templates/kuttl/smoke/10-install-opa.yaml.j2 [hunk 5]
```diff
@@ -33,6 +33,12 @@ spec:
     config:
       logging:
         enableVectorAgent: {{ lookup('env', 'VECTOR_AGGREGATOR') | length > 0 }}
+      resources:
+        cpu:
+          max: '1'
+          min: '300m'
+        memory:
+          limit: '512Mi'
```
Comment (sbernauer) [inline tests/templates/kuttl/smoke/10-install-opa.yaml.j2:41]: Can we test the default values here to make sure they are enough to pass the smoke tests?

**Review by sbernauer — COMMENTED** [review]
Thanks!

**Comment by maltesander** [comment]
test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/opa-operator-it-custom/67/

File: tests/templates/kuttl/smoke/10-install-opa.yaml.j2 (hunk 5 above)
Comment (maltesander) [reply tests/templates/kuttl/smoke/10-install-opa.yaml.j2]: https://github.com/stackabletech/opa-operator/pull/453/commits/0bb646390e50a01908b7097eddec8b4fc78af27a

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**